numismatic/feeds/base.py

```diff
@@ -139,6 +146,85 @@ class RestClient(abc.ABC):
     cache_dir = attr.ib(default=None)
     requester = attr.ib(default='base')
 
+    def listen(self, symbol, channel, interval=1.0, exchange=None):
+        exchange = exchange if exchange else self.exchange
+        channel_name = f'{exchange}--{symbol}--{channel.__name__}'
+        logger.info(f'Subscribing to {channel_name} ...')
+        # timer
+
+        # FIXME: get the symbol properly
+        # application
+        def _get_raw_channel():
+            messages = channel(symbol[:3], symbol[3:], exchange=exchange,
+                               raw=True)
+            packet = '\n'.join(json.dumps(msg) for msg in messages)
+            return packet
+
+        channel_info = {'channel': channel.__name__}
+        subscription = Subscription(exchange=exchange,
+                                    symbol=symbol,
+                                    channel='ticker',
+                                    channel_info=channel_info, 
+                                    client=self,
+                                    listener=None,
+                                    handlers=self._get_handlers(),
+                                    )
+        # FIXME: find a better way to do this
+        subscription.listener = self._listener(subscription, interval=interval,
```

Comment:
Can you ping me on WA and detail ideas about what is a better way to do this?

Reply:
I'm not sure. I think it will come out in the wash once we think carefully about how to share the sockets across subscriptions.